src/NUnitFramework/tests/Assertions/AssertFailTests.cs

```diff
@@ -125,5 +125,29 @@ public void CatchingAssertionExceptionMakesTestPass()
             // Ensure that no spurious info was recorded from the assertion
             Assert.That(TestExecutionContext.CurrentContext.CurrentResult.AssertionResults.Count, Is.EqualTo(0));
         }
+
+        [Test]
+        public void AssertCatchMakesTestPass()
+        {
+            Assert.Catch(() =>
+            {
+                Assert.Fail("This should not be seen");
+            });
+
+            // Ensure that no spurious info was recorded from the assertion
+            Assert.That(TestExecutionContext.CurrentContext.CurrentResult.AssertionResults.Count, Is.EqualTo(0));
```

Comment:
Can we use `IsEmpty` instead of checking the `Count` property directly? We have an assertion for empty collections right?

Reply:
I thought of it as "number of assertion results is zero" and wrote it that way. Style, I guess.

Reply:
> I thought of it as "number of assertion results is zero" and wrote it that way. Style, I guess.

On this particular one, I disagree that it is a matter of style. The collection constraints provide better semantic and were created for this exact purpose. Additionally, they also help with reducing the amount of child member access/law of demeter violation (which is already insane over here with 4 levels).

Reply:
@julealgon I favor your approach because it yields a clearer failure message, but I think this is a matter of style no different than choosing between `collection.Count == 0` and `collection.Any()`.

Reply:
@jnm2 Sold! Based on the failure message.

I still hold back on a lot of Linq stuff for the same reason you like it. It just doesn't speak to me. I have to think twice to remember what each method does, whereas I know immediately what count == 0 means. That's particularly true with a method like Any() - a name that has its own use in NUnit. 

I wish we had a bit more specialization. Then, the people who did most of the work in a particular area of the code could decide on stylistic issues to their liking. Those who only dabbled would use that style for consistency. Of course, I was the guy for a long time in this particular area... 😄

Reply:
Ironically, contrary to what I just said, I still instinctively use `.Count == 0` rather than `.Any()`. I'm still trying to kick the habit as a premature optimization.